doc/table.dox

```diff
@@ -158,17 +158,58 @@ namespace lsst { namespace afw { namespace table {
  *  discussed below).
  *
  *  @section afwTableFieldNames Field Names
- *  By convention, field names are all lowercase and have '.'-separated elements.  Only letters, numbers
- *  and periods should be used.  These rules are not enforced, but names that do not meet these requirements
- *  may not round-trip correctly in FITS (periods are converted to underscores in the FITS persistence
- *  layer, so we cannot distinguish between the two when we read tables from FITS).
- *  Schema provides extra functionality for names with period-separated elements; these elements can
- *  be accessed separately individually with the bracket operators.  More information on schema namespaces
- *  can be found in the Schema and SubSchema class documentation, and the testSchema.py unit test may
- *  also be a useful example.
- *
- *  Other field strings (documentation and units) are essentially arbitrary, but should not contain
- *  single quotes, as these may also confuse FITS parsers (even when escaped).
+ *
+ *  By convention, field names are camel case and have '_'-separated elements.
+ *  Only letters, numbers, and underscores should be used.
+ *  These rules are not enforced, but names that do not meet these requirements
+ *  may not round-trip correctly in FITS.
+ *
+ *  Underscores should be used as a sort of namespace separator (much like '.'
+ *  in Python; we use underscore instead so we don't need to translate field
+ *  names when using them in SQL).  These namespace elements typically indicate
+ *  things like the module or Task containing the algorithm that produced the
+ *  field, the name of the algorithm, and (finally) the name of the particular
+ *  output produced by that algorithm.
+ *
+ *  For example:
+ *
+ *    - `base_SdssShape_xx` is produced by the SdssShape algorithm located in
+ *      the meas_base package (we drop "meas_" since so many algorithms are in
+ *      meas_* packages; see `lsst.meas.base.generateAlgorithmName`), and this
+ *      particular field represents the x-x second-moment of the source.
+ *
+ *    - `deblend_nChild` is produced by `SourceDeblendTask`, where `deblend`
+ *      is an abbreviation of the Task, and `nChild` is a value produced by
+ *      it.  There are no hard rules for how to abbreviate the name of a Task
+ *      when generating its field names; we trust developer judgement in
+ *      selecting a prefix that is both concise and unambiguous.
+ *
```

Comment:
Actually, could I ask that you include an example from one of the cases that pointed this issue out (just to be overly complete!)?

Reply:
Which cases do you mean?  The ones on DM-14997?  I'm vaguely aware that different Tasks do different things, which is why I've avoiding trying to declare a rule that I know we don't follow, but I haven't actually thought about which ones use good names and why.

Reply:
Yeah, like the calib_photometryUsed -> calib_photometry_used, for example.  A call has to be made there to homogenize (or do you think that requires an RFC.  I liked your <module-or-task>_<algorithm>_<state> argument...)

Reply:
I think there will need to be an RFC regardless, because changing the schemas produced by the pipeline breaks any code that reads those outputs; it's like an API change.

I actually think the reason to prefer `calib_photometry_used` over `calib_photometryUsed` (now that I think about it) is that you want underscores when you want the prefix to be considered a conceptual group.  I'll go ahead and add that rule of thumb as well.

Reply:
Ok, shall I write an RFC for which DM-14997 will be the implementation ticket?

Reply:
> Ok, shall I write an RFC for which DM-14997 will be the implementation ticket?

Yup, that sounds perfect.